python/ray/tune/trial.py

```diff
@@ -224,14 +224,30 @@ def __init__(self,
         if trainable_cls:
             default_resources = trainable_cls.default_resource_request(
                 self.config)
+
+            # If Trainable returns resources, do not allow manual override via
+            # `resources_per_trial` by the user.
             if default_resources:
                 if resources:
                     raise ValueError(
                         "Resources for {} have been automatically set to {} "
                         "by its `default_resource_request()` method. Please "
                         "clear the `resources_per_trial` option.".format(
                             trainable_cls, default_resources))
-                resources = default_resources
+
+                # New way: Trainable returns a PlacementGroupFactory object.
+                if isinstance(default_resources, PlacementGroupFactory):
+                    # If Trainable returns placement group factory, only use it
+                    # if no manual `placement_group_factory` are has been provided
+                    # by user.
```

Comment:
Ah so this means that users could overwrite default resource requests with their own placement groups. Is this what we want here? Old behavior was just to disallow any `resources_per_trial` when a default resource request exists.

```suggestion
                    # If Trainable returns placement group factory, only use it
                    # if no manual `placement_group_factory` has been provided
                    # by user.
```